cmd/osm-controller/osm-controller_test.go

```diff
@@ -2,74 +2,136 @@ package main
-		})
-	})
-})
-
-var _ = Describe("Test joining of URL paths", func() {
-	It("should correctly join URL paths", func() {
-		final := joinURL("http://foo", "/bar")
-		Expect(final).To(Equal("http://foo/bar"))
-	})
-
-	It("should correctly join URL paths", func() {
-		final := joinURL("http://foo/", "/bar")
-		Expect(final).To(Equal("http://foo/bar"))
-	})
-
-	It("should correctly join URL paths", func() {
-		final := joinURL("http://foo/", "bar")
-		Expect(final).To(Equal("http://foo/bar"))
-	})
-
-	It("should correctly join URL paths", func() {
-		final := joinURL("http://foo", "bar")
-		Expect(final).To(Equal("http://foo/bar"))
-	})
-})
+func TestConfigureDebugServer(t *testing.T) {
+	assert := assert.New(t)
+	const testPort = 9999
+	const validRoutePath = "/debug/test1"
+
+	defaultConfigMap := map[string]string{
+		"enabled_debug_server": "true",
+	}
+	kubeClient := testclient.NewSimpleClientset()
+	stop := make(chan struct{})
+	osmNamespace := "-test-osm-namespace-"
+	osmConfigMapName := "-test-osm-config-map-"
+	cfg := configurator.NewConfigurator(kubeClient, stop, osmNamespace, osmConfigMapName)
+	configMap := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: osmNamespace,
+			Name:      osmConfigMapName,
+		},
+		Data: defaultConfigMap,
+	}
+	_, err := kubeClient.CoreV1().ConfigMaps(osmNamespace).Create(context.TODO(), &configMap, metav1.CreateOptions{})
+	assert.Nil(err)
+
+	mockCtrl := gomock.NewController(t)
+	mockDebugServer := debugger.NewMockDebugServer(mockCtrl)
+	mockDebugServer.EXPECT().GetHandlers().Return(map[string]http.Handler{
+		validRoutePath: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}),
+	}).AnyTimes()
+
+	type configDebugServerTest struct {
+		defaultEnableDebug bool
+		enableDebug        string
+		serverStopped      bool
+	}
+
+	debugServer := httpserver.NewDebugHTTPServer(mockDebugServer, testPort)
+
+	configDebugServerTests := []configDebugServerTest{
+		{true, "false", true},  //stop debug server
+		{false, "true", false}, //start debug server
+	}
+	errCh := make(chan error)
+
+	for _, cdst := range configDebugServerTests {
+		defaultConfigMap["enable_debug_server"] = cdst.enableDebug
+		configMap := v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: osmNamespace,
+				Name:      osmConfigMapName,
+			},
+			Data: defaultConfigMap,
+		}
+		debugServerRunning := cdst.defaultEnableDebug
+
+		go configureDebugServer(debugServer, mockDebugServer, debugServerRunning, cfg, errCh)
+		_, err = kubeClient.CoreV1().ConfigMaps(osmNamespace).Update(context.TODO(), &configMap, metav1.UpdateOptions{})
+		assert.Nil(err)
+
+		close(stop)
+		time.Sleep(2 * time.Second)
+		if cdst.serverStopped {
+			//Checks that debug server is closed
+			assert.Equal(debugServer.Server.ListenAndServe(), http.ErrServerClosed)
+		}
+
+		stop = make(chan struct{})
+		cfg = configurator.NewConfigurator(kubeClient, stop, osmNamespace, osmConfigMapName)
```

Comment:
duplicate I think

Reply:
@michelleN we have to make a new configurator because we close the stop channel in L147 so the old configurator also stopped